### PR TITLE
Add hwdb parser to check for inconsistencies

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: hwdb/60-evdev.hwdb [hunk 1]
```diff
@@ -76,9 +76,6 @@ evdev:input:b0003v05ACp0254*
  EVDEV_ABS_36=::92
 
 # MacbookPro10,1 (unibody, June 2012)
-evdev:input:b0003v05ACp0259*
-evdev:input:b0003v05ACp025A*
-evdev:input:b0003v05ACp025B*
```
Comment (keszybz) [inline hwdb/60-evdev.hwdb:81]: @whot You added this one, please have a look.

File: hwdb/60-keyboard.hwdb [hunk 2]
```diff
@@ -888,7 +888,7 @@ evdev:atkbd:dmi:bvn*:bvr*:bd*:svnOLPC:pnXO:*
  KEYBOARD_KEY_c2=f8
  KEYBOARD_KEY_c3=f9
  KEYBOARD_KEY_c4=f10
- KEYBOARD_KEY_c7=f11
+ # KEYBOARD_KEY_c7=f11 # FIXME!
```
Comment (keszybz) [inline hwdb/60-keyboard.hwdb:891]: Somebody with an OLPC should check this.

File: hwdb/70-mouse.hwdb [hunk 3]
```diff
@@ -342,8 +342,6 @@ mouse:usb:v046dp402d:name:Logitech M560:
 mouse:usb:v046dpc52b:name:Logitech Unifying Device. Wireless PID:402d:
  MOUSE_DPI=1000@125
 
-# Logitech V220 Cordless Optical Mouse
-mouse:usb:v046dpc51b:name:Logitech USB Receiver:
```
Comment (keszybz) [inline hwdb/70-mouse.hwdb:346]: @phomes You added those, please have a look.

**Comment by poettering** [comment]
lgtm. But maybe @whot and @martinpitt should bless this too! @whot? @martinpitt?

File: hwdb/60-evdev.hwdb (hunk 1 above)
Comment (whot) [inline hwdb/60-evdev.hwdb:81]: these should be fixed (range is 0262, 0263 and 0264), it's a simple copy-paste error. I sent a PR to fix these here: https://github.com/systemd/systemd/pull/3921

File: hwdb/70-mouse.hwdb (hunk 3 above)
Comment (whot) [inline hwdb/70-mouse.hwdb:346]: c51b seems to be a unifying receiver PID. For the c52b ID we have the kernel extract the actual device and present it as such, @bentiss do you know what the deal is where for the c51b? I can't find it in the kernel sources, is this a different model?

Comment (bentiss) [inline hwdb/70-mouse.hwdb:346]: According to http://support.logitech.com/en_us/product/v220-cordless-optical-mouse-for-notebooks it seems the mouse is using a pre-version of the small unifying receiver we know now. If there are 2 mice with the same receiver, that means that the values should both be dropped IMO.

I can't find in the docs if c51b would be compatible with hid-logitech-dj.ko, but I am not sure either if it is worth working on those old HW.

**Comment by keszybz** [comment]
PR reopened, I still hope to run those tests automatically.

I removed the commit for Apple touchpads, since it was resolved in #3291, and dropped both entries for the unifying receiver as suggested by @bentiss.

**Comment by keszybz** [comment]
Hm, it seems that github does not allow reopening PRs. I'll make a new one.

**Pull request closed without merging**